Sources/SwiftSyntaxBuilderGeneration/SyntaxBuildableType.swift

```diff
@@ -68,81 +66,98 @@ struct SyntaxBuildableType: Hashable {
     syntaxKind
   }
 
-  /// Return the name of the `Buildable` type that is the main entry point for building
+  /// Return the `Buildable` type that is the main entry point for building
   /// SwiftSyntax trees using `SwiftSyntaxBuilder`.
   ///
   /// These names look as follows:
   ///  - For nodes: The node name, e.g. `IdentifierExpr` (these are implemented as structs)
   ///  - For base kinds: `<BaseKind>Buildable`, e.g. `ExprBuildable` (these are implemented as protocols)
   ///  - For token: `TokenSyntax` (tokens don't have a dedicated type in SwiftSyntaxBuilder)
-  /// If the type is optional, this terminates with a '?'.
-  var buildable: String {
-    if isToken {
-      // Tokens don't have a dedicated buildable type.
-      return "TokenSyntax\(optionalQuestionMark)"
-    } else if SYNTAX_BASE_KINDS.contains(syntaxKind) {
-      return "\(syntaxKind)Buildable\(optionalQuestionMark)"
-    } else {
-      return "\(syntaxKind)\(optionalQuestionMark)"
-    }
+  /// If the type is optional, the type is wrapped in an `OptionalType`.
+  var buildable: ExpressibleAsTypeBuildable {
+    optionalWrapped(type: buildableBaseName)
   }
 
   /// Whether parameters of this type should be initializable by a result builder.
   /// Used for certain syntax collections and block-like structures (e.g. `CodeBlock`,
   /// `MemberDeclList`).
   var isBuilderInitializable: Bool {
-    BUILDER_INITIALIZABLE_TYPES.keys.contains(nonOptional.buildable)
+    BUILDER_INITIALIZABLE_TYPES.keys.contains(buildableBaseName)
   }
 
   /// A type suitable for initializing this type through a result builder (e.g.
   /// returns `CodeBlockItemList` for `CodeBlock`) and otherwise itself.
   var builderInitializableType: Self {
-    let buildable = nonOptional.buildable
-    return Self(
-      syntaxKind: BUILDER_INITIALIZABLE_TYPES[buildable].flatMap { $0 } ?? buildable,
+    Self(
+      syntaxKind: BUILDER_INITIALIZABLE_TYPES[buildableBaseName].flatMap { $0 } ?? buildableBaseName,
       isOptional: isOptional
     )
   }
 
   /// The type from `buildable()` without any question marks attached.
   /// This is used for the `create*` methods defined in the `ExpressibleAs*` protocols.
   var buildableBaseName: String {
-    nonOptional.buildable
+    if isToken {
+      // Tokens don't have a dedicated buildable type.
+      return "TokenSyntax"
+    } else if SYNTAX_BASE_KINDS.contains(syntaxKind) {
+      return "\(syntaxKind)Buildable"
+    } else {
+      return syntaxKind
+    }
   }
 
-  /// The `ExpressibleAs*` Swift type for this syntax kind. Tokens don't
-  /// have an `ExpressibleAs*` type, so for those this method just returns
-  /// `TokenSyntax`. If the type is optional, this terminates with a `?`.
-  var expressibleAs: String {
+  /// The `ExpressibleAs*` Swift type for this syntax kind without any
+  /// question marks attached.
+  var expressibleAsBaseName: String {
```

Comment:
This PR factors out a few `...BaseName` properties (`expressibleAs`, `syntax`, `buildable`). These represent a non-optional type and will still be strings (as opposed to, e.g. `ExpressibleAsTypeBuildable`s) to make them easy to embed in string interpolations and across type and expression contexts.

Thus `.xyzBaseName` replaces the earlier `.nonOptional.xyz` idiom.